src/classes/ALLO_AllocationsService_TEST.cls

```diff
@@ -0,0 +1,853 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification,Then w are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Allocations
+* @group-content ../../ApexDocContent/Allocations.htm
+* @description Tests for ALLO_AllocationsService class
+*/
+@IsTest(isParallel=true)
+public class ALLO_AllocationsService_TEST {
+    /**
+     * @description Executes various Opportunity and Payment Allocation Scenarios when
+     *              Payment and Default Allocations are enabled
+     *              [ALLO_AllocationsService.processRecords]
+     *
+     */
+    @isTest
+    private static void confirmProcessRecordsUpdatesAllocationsCorrectlyWhenPaymentAllocationsAreEnabled() {
+        ALLO_UnitTestHelper_TEST.AllocationSetupTest alloSetupTest = new ALLO_UnitTestHelper_TEST.AllocationSetupTest()
```

Comment:
This is a 280 line test method. Ideally it should test one concept at a time. I'm not necessarily able to figure out if it does or not. Is there a way to break this test into more logical chunks that test individual blocks of functionality?

Reply:
So...this test, and the others like it are meant to confirm that operations being done in a batch iteration, or multiple records being updated in a trigger context doing different things all come together and properly render the correct allocations.  That is why it is so long.  Testing serially each of the allocations scenarios would be very easy to write, but doesn't really cover a bulkified scenario.  I am already testing each of the individual services that this calls.  It is intended to make sure that the data sent in comes back out properly, depending on the configuration.